contracts/flow/token/BloctoPass.cdc

```diff
@@ -294,87 +431,117 @@ pub contract BloctoPass: NonFungibleToken {
+        // and returns it to the caller as a reference to the NFT
+        access(CollectionPrivateEntitlement)
+        fun borrowWithdraw(id: UInt64): auth(FungibleToken.Withdraw) &BloctoPass.NFT {
+            let bloctoPassRef: auth(FungibleToken.Withdraw) &{NonFungibleToken.NFT} = (&self.ownedNFTs[id])!
+            return bloctoPassRef as! auth(FungibleToken.Withdraw) &BloctoPass.NFT
+        }
+        
+        access(all)
+        view fun getSupportedNFTTypes(): {Type: Bool} {
+            let supportedTypes: {Type: Bool} = {}
+            supportedTypes[Type<@BloctoPass.NFT>()] = true
+            return supportedTypes
+        }
+        
+        access(all)
+        view fun isSupportedNFTType(type: Type): Bool {
+            return type == Type<@BloctoPass.NFT>()
+        }
+        
+        access(all)
+        fun createEmptyCollection(): @{NonFungibleToken.Collection} {
+            return <-create Collection()
         }
     }
 
     // public function that anyone can call to create a new empty collection
-    pub fun createEmptyCollection(): @NonFungibleToken.Collection {
-        return <- create Collection()
+    access(all)
+    fun createEmptyCollection(nftType: Type): @{NonFungibleToken.Collection} {
+        return <-create Collection()
     }
 
-    pub resource interface MinterPublic {
-        pub fun mintBasicNFT(recipient: &{NonFungibleToken.CollectionPublic})
+    access(all)
+    resource interface MinterPublic {
+        access(all)
+        fun mintBasicNFT(recipient: &{NonFungibleToken.CollectionPublic}): Void
     }
 
     // Resource that an admin or something similar would own to be
     // able to mint new NFTs
     //
-    pub resource NFTMinter: MinterPublic {
+    access(all)
+    resource NFTMinter: MinterPublic {
 
         // adds a new predefined lockup schedule
-        pub fun setupPredefinedLockupSchedule(lockupSchedule: {UFix64: UFix64}) {
+        access(NFTMinterEntitlement)
+        fun setupPredefinedLockupSchedule(lockupSchedule: {UFix64: UFix64}) {
             BloctoPass.predefinedLockupSchedules.append(lockupSchedule)
-
             emit LockupScheduleDefined(id: BloctoPass.predefinedLockupSchedules.length, lockupSchedule: lockupSchedule)
         }
 
         // updates a predefined lockup schedule
         // note that this function should be avoided 
-        pub fun updatePredefinedLockupSchedule(id: Int, lockupSchedule: {UFix64: UFix64}) {
+        access(NFTMinterEntitlement)
+        fun updatePredefinedLockupSchedule(id: Int, lockupSchedule: {UFix64: UFix64}) {
             BloctoPass.predefinedLockupSchedules[id] = lockupSchedule
-
             emit LockupScheduleUpdated(id: id, lockupSchedule: lockupSchedule)
         }
 
         // mintBasicNFT mints a new NFT without any special metadata or lockups
-        pub fun mintBasicNFT(recipient: &{NonFungibleToken.CollectionPublic}) {
-            self.mintNFT(recipient: recipient, metadata: {})
+        access(all)
```

Comment:
Should this have an entitlement?

Reply:
thanks for your comment,
access(all) may correct, because:

1. Original interface code 
```
 pub resource interface MinterPublic {
        pub fun mintBasicNFT(recipient: &{NonFungibleToken.CollectionPublic})
    }
```
it wants any user can mint it because "Public"


2. double checking with the transactions/staking/app/EnableBltStake.cdc 
(the script using from iOS app, and NONE update code)
(the 0x0f9df91c9121c460 is mainnet BloctoPass contract account)

it needs any user can mint the the NFT and use it to stake BLT token 
```
    if collectionRef.getIDs().length == 0 {
      let minterRef = getAccount(0x0f9df91c9121c460).getCapability(/public/bloctoPassMinter)
        .borrow<&{BloctoPass.MinterPublic}>()
        ?? panic("Could not borrow minter public reference")

      minterRef.mintBasicNFT(recipient: collectionRef) 
    }
```
 
 if it's not access(all), it will be failed.